src/utils/fileManager.ts

```diff
@@ -0,0 +1,17 @@
+import fs from 'fs';
+
+export const WriteToXmlFile = async (fileName: string, datToBeWritten: string) => {
+    const path = 'src/GeneratedXmlFiles';
+    if (!(fs.existsSync(path))) {
+        fs.mkdirSync(path);
+    }
+
+    try {
+        const logger = fs.createWriteStream(`${path}/${fileName}.xml`, {
```

Comment:
You're creating a stream here but flushing all the writing at once. You also forgot to close the stream or wait for it to flush.
It's more appropriate to use `writeFileSync` here.